pkg/graveler/graveler.go

```diff
@@ -1044,8 +1101,8 @@ func (g *Graveler) List(ctx context.Context, repositoryID RepositoryID, ref Ref)
 	if err != nil {
 		return nil, err
 	}
-	if reference.Type() == ReferenceTypeBranch {
-		stagingList, err := g.StagingManager.List(ctx, reference.Branch().StagingToken)
+	if reference.Type == ReferenceTypeBranch {
+		stagingList, err := g.StagingManager.List(ctx, reference.StagingToken)
```

Comment:
So if we explicitly use "@" to get only a commit - we get empty staging token here. And then you rely on the query from the DB to return empty?
If so, I would explicitly check for an empty staging token here, and not call the staging manager at all.

This comment may be relevant at other places in this file

Reply:
true, I'll switch the check to match ResolvedBranchModifierStaging and not the type of the reference we resolved